### PR TITLE
[Core] Remove custom framework API

### PR DESCRIPTION
**Pull request by mhutch**
It's not useful to be able to define custom frameworks
inside MD, only MSBuild frameworks actually work now.

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/TargetFramework.cs [hunk 1]
```diff
@@ -79,10 +68,6 @@ internal TargetFramework (TargetFrameworkMoniker id)
 			Assemblies = new AssemblyInfo[0];
 		}
 		
-		public bool Hidden {
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/TargetFramework.cs]: Just for the sake of not breaking the API, can you make it just return true?

Comment (mhutch) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/TargetFramework.cs]: or false :)

Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/TargetFramework.cs]: Yeah, false :)

**Comment by mhutch** [comment]
@slluis i addressed the feedback

**Comment by mhutch** [comment]
@slluis updated!

**Comment by mhutch** [comment]
@slluis rebased again

**Pull request merged**